qa/tasks/dnsmasq.py

```diff
@@ -83,7 +83,7 @@ def setup_dnsmasq(remote, testdir, cnames):
     # restart dnsmasq
     remote.run(args=['sudo', 'systemctl', 'restart', 'dnsmasq'])
     # verify dns name is set
-    remote.run(args=['ping', '-c', '4', cnames.keys()[0]])
+    remote.run(args=['ping', '-c', '4', next(iter(cnames.keys()))])
```

Comment:
(nit) Hm, wonder why the code is checking only first address, can we fix it here: to cycle throw all cnames?

Reply:
this is not in the scope of this PR, i'd prefer deferring it after we get this change merged and having it in a following-up.